src/ops/fuelup_check.rs

```diff
@@ -0,0 +1,146 @@
+
+    let cfg = Config::from_env()?;
+    let toolchains = cfg.list_toolchains()?;
+    let mut latest_versions: HashMap<String, Version> = HashMap::new();
+    let components = Vec::from([component::FORC, component::FUEL_CORE, component::FUELUP]);
+
+    for component in components {
+        let download_cfg: DownloadCfg = DownloadCfg::new(component, None)?;
+        latest_versions.insert(component.to_string(), download_cfg.version);
+    }
+
+    for toolchain in toolchains {
+        if let Ok(toolchain) = Toolchain::from(&toolchain) {
+            bold(|s| writeln!(s, "{}", &toolchain.name));
+            for component in [component::FORC, component::FUEL_CORE] {
+                let component_executable = toolchain.path.join(component);
+
+                match std::process::Command::new(&component_executable)
+                    .arg("--version")
+                    .output()
+                {
+                    Ok(o) => {
+                        let version = Version::parse(
+                            String::from_utf8_lossy(&o.stdout)
+                                .split_whitespace()
+                                .collect::<Vec<&str>>()[1],
+                        )?;
+
+                        bold(|s| write!(s, "  {} - ", &component));
+                        if version == latest_versions[component] {
+                            with_color_maybe_bold(|s| write!(s, "Up to date"), Color::Green, true);
+                            println!(" : {}", version);
+                        } else {
+                            with_color_maybe_bold(
+                                |s| write!(s, "Update available"),
+                                Color::Yellow,
+                                true,
+                            );
+                            println!(" : {} -> {}", version, latest_versions[component]);
+                        }
+                    }
+                    Err(e) => {
+                        print!("  ");
+                        bold(|s| write!(s, "{}", &component));
+                        print!(" - ");
+                        if component_executable.exists() {
+                            println!("execution error - {}", e);
+                        } else {
+                            println!("not found");
+                        }
+                    }
+                };
+
+                if verbose && component == component::FORC {
+                    for plugin in SUPPORTED_PLUGINS {
+                        check_plugin(&toolchain, plugin, &latest_versions[component::FORC])?;
+                    }
+                }
+            }
+        }
+    }
+
+    match std::process::Command::new(component::FUELUP)
+        .arg("--version")
+        .output()
+    {
+        Ok(o) => {
+            let version = Version::parse(
+                String::from_utf8_lossy(&o.stdout)
+                    .split_whitespace()
+                    .collect::<Vec<&str>>()[1],
```

Comment:
If we're only using the 2nd element, no need to collect the whole output into a `Vec`. It would be better to do something like:

```rust
let version = Version::parse(
    String::from_utf8_lossy(&o.stdout)
        .split_whitespace()
        .nth(1)
        .expect("expected version")
    )
);